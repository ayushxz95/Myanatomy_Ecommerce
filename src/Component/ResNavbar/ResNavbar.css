.menu-icon {
    margin-left: 10px;
    padding: 4px;
}
.box{
    display: block;
    text-align: left;
    padding: 14px 16px;
    text-decoration: none; 
}
.box:hover{
    background-color: rgb(133, 128, 128);
    cursor: pointer;
  }
  .box-search{
    margin-top: 8px;
    width: 100px;
    height:30px;
    border-radius: 30px;
    display:flex;
    flex-direction: row;
    background-color: white;
    padding: 4px;
    margin-left: 10px;

  }
  .box-search > i{
    padding: 2px;
    font-size: 20px;
    color: #777;
  }
  .box-search > input{
    flex:1;
    border:none;
    outline: none;
  }