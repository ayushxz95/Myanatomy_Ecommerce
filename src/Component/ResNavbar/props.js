import {
    fetchCategoryItemsSuccess,
    fetchItems,
    searchItem,
  } from "../../Redux/Actions/ItemAction";

const mapDispatchToProps = (dispatch) => {
    return {
      fetchCategoryItems: (category) =>
        dispatch(fetchCategoryItemsSuccess(category)),
      fetchItems: () => dispatch(fetchItems()),
      searchItem: (text) => dispatch(searchItem(text)),
    };
  };

  export {mapDispatchToProps}