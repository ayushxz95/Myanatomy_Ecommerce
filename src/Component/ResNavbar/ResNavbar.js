import React, { Component } from "react";
import { Drawer, IconButton, Typography, Box } from "@mui/material";
import { AiOutlineMenu } from "react-icons/ai";
import { GoSearch } from "react-icons/go";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";
import {
  fetchCategoryItemsSuccess,
  fetchItems,
  searchItem,
} from "../../Redux/Actions/ItemAction";
import "./ResNavbar.css";

class ResNavbar extends Component {
  constructor() {
    super();
    this.state = {
      isDrawerOpen: false,
    };
  }
  handelDrawer = () => {
    this.setState((prev) => ({ isDrawerOpen: !prev.isDrawerOpen }));
  };
  onChange = (e) => {
    this.props.searchItem(e.target.value);
  };
  render() {
    return (
      <div>
        <IconButton
          size="large"
          edge="start"
          color="inherit"
          aria-label="logo"
          onClick={this.handelDrawer}
        >
          <AiOutlineMenu color="white" className="menu-icon" />
        </IconButton>
        <Drawer
          anchor="left"
          open={this.state.isDrawerOpen}
          onClose={this.handelDrawer}
        >
          <Box p={2} width="250px" textAlign="center" role="presentation">
            <Typography variant="h5" component="div">
              Menu
            </Typography>
            <div className="menu-items">
              <div>
                <a
                  className="box"
                  onClick={() => {
                    this.props.fetchItems();
                    this.handelDrawer();
                  }}
                >
                  Home
                </a>
              </div>
              <div>
                <a
                  className="box"
                  onClick={() => {
                    this.props.fetchCategoryItems("Cloths");
                    this.handelDrawer();
                  }}
                >
                  Cloths
                </a>
              </div>
              <div>
                <a
                  className="box"
                  onClick={() => {
                    this.props.fetchCategoryItems("Electronics");
                    this.handelDrawer();
                  }}
                >
                  Electronics
                </a>
              </div>
              <div>
                <a
                  className="box"
                  onClick={() => {
                    this.props.fetchCategoryItems("Grocery");
                    this.handelDrawer();
                  }}
                >
                  Grocery
                </a>
                <a className="box-search">
                  <i>
                    <GoSearch color="#ef257a"/>
                  </i>
                  <input
                    className="search-input"
                    type="text"
                    onChange={this.onChange}
                    placeholder=" Search..."
                  ></input>
                </a>
              </div>
            </div>
          </Box>
        </Drawer>
      </div>
    );
  }
}

// const mapDispatchToProps = (dispatch) => {
//   return {
//     fetchCategoryItems: (category) =>
//       dispatch(fetchCategoryItemsSuccess(category)),
//     fetchItems: () => dispatch(fetchItems()),
//     searchItem: (text) => dispatch(searchItem(text)),
//   };
// };

ResNavbar.propTypes = {
  fetchCategoryItems: PropTypes.func,
  fetchItems: PropTypes.func,
  searchItem: PropTypes.func,
};

export default ResNavbar;
