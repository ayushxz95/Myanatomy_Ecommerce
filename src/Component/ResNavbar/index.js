import { connect } from "react-redux";
import ResNavbar from "./ResNavbar";
import { mapDispatchToProps } from "./props";



export default connect(null, mapDispatchToProps)(ResNavbar);