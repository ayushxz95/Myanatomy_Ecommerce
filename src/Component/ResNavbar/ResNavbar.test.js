import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import ResNavbar from "./ResNavbar";

configure({ adapter: new Adapter() });


describe("Testing ResNavbar Component", () => {
  it("Matching ResNavbar's snapshot", () => {
    const component = shallow(
      <ResNavbar />
    );
    expect(component).toMatchSnapshot();
  });
});