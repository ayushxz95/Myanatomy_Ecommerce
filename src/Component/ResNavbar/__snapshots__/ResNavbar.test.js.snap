// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ResNavbar Component Matching ResNavbar's snapshot 1`] = `
<div>
  <ForwardRef(IconButton)
    aria-label="logo"
    color="inherit"
    edge="start"
    onClick={[Function]}
    size="large"
  >
    <AiOutlineMenu
      className="menu-icon"
      color="white"
    />
  </ForwardRef(IconButton)>
  <ForwardRef(Drawer)
    anchor="left"
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(Box)
      p={2}
      role="presentation"
      textAlign="center"
      width="250px"
    >
      <ForwardRef(Typography)
        component="div"
        variant="h5"
      >
        Menu
      </ForwardRef(Typography)>
      <div
        className="menu-items"
      >
        <div>
          <a
            className="box"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div>
          <a
            className="box"
            onClick={[Function]}
          >
            Cloths
          </a>
        </div>
        <div>
          <a
            className="box"
            onClick={[Function]}
          >
            Electronics
          </a>
        </div>
        <div>
          <a
            className="box"
            onClick={[Function]}
          >
            Grocery
          </a>
          <a
            className="box-search"
          >
            <i>
              <GoSearch
                color="#ef257a"
              />
            </i>
            <input
              className="search-input"
              onChange={[Function]}
              placeholder=" Search..."
              type="text"
            />
          </a>
        </div>
      </div>
    </ForwardRef(Box)>
  </ForwardRef(Drawer)>
</div>
`;
